tools/licenses/lib/main.dart

```diff
@@ -2205,6 +2205,38 @@ class RepositoryGarnetFidlDirectory extends RepositoryDirectory {
   }
 }
 
+class RepositoryTopazDirectory extends RepositoryDirectory {
+  RepositoryTopazDirectory(RepositoryDirectory parent, fs.Directory io) : super(parent, io);
+
+  @override
+  bool shouldRecurse(fs.IoNode entry) {
+    return entry.name == 'lib'
+        && super.shouldRecurse(entry);
```

Comment:
I'd feel much more comfortable with a black-list than a white-list here (and below). A white-list risks that we add code to a new directory and never notice. Worst case with a black list we end up with an extra license. Worst case with a white list we violate a license.

Reply:
Whitelist `topaz/lib/tonic` is now blacklist `topaz/tools` and `topaz/shell/third_party/QR-Code-generator`